examples/user_guide/17-Dashboards.ipynb

```diff
@@ -117,15 +117,45 @@
    "outputs": [],
    "source": [
     "# Apply rolling mean\n",
-    "window = Params(explorer, ['rolling_window'])\n",
-    "smoothed = rolling(stock_dmap, streams=[window])\n",
+    "smoothed = rolling(stock_dmap, rolling_window=explorer.param.rolling_window)\n",
     "\n",
     "# Find outliers\n",
-    "outliers = rolling_outlier_std(stock_dmap, streams=[window]).opts(\n",
-    "    hv.opts.Scatter(color='red', marker='triangle')\n",
-    ")\n",
+    "outliers = rolling_outlier_std(stock_dmap, rolling_window=explorer.param.rolling_window).opts(\n",
+    "    color='red', marker='triangle')\n",
+    "\n",
+    "pn.Row(explorer.param, (smoothed * outliers).opts(width=600, padding=0.1))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## A function based approach\n",
+    "\n",
+    "Instead of defining a whole Parameterized class we can also use the ``depends`` decorator to directly link the widgets to a DynamicMap callback function. This approach makes the link between the widgets and the computation very explicit at the cost of tying the widget and display code very closely together.\n",
+    "\n",
+    "Instead of declaring the dependencies as strings we map the parameter instance to a particular keyword argument in the ``depends`` call. In this way we can link the symbol to the ``RadioButtonGroup`` value and the ``variable`` to the ``Select`` widget value:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "symbol = pn.widgets.RadioButtonGroup(options=stock_symbols)\n",
+    "variable = pn.widgets.Select(options=variables)\n",
+    "window = pn.widgets.IntSlider(name='Rolling Window', value=10, start=1, end=365)\n",
```

Comment:
`rolling_window` for consistency with the earlier parameter name?

Reply:
No the other one is a parameter name, this is effectively the label. Somewhat regret using name for this in panel widgets but stuck with it now.

Reply:
Not quite sure what you mean. I'm a little confused as the handles `symbol` and `variable` here do match the `symbol` and `variable` parameters earlier in the class definition. Why does `window` need to be inconsistent?

Reply:
Ah, you mean the variable name, not the ``name`` parameter.